src/test/java/uk/ac/ebi/eva/pipeline/parameters/validation/step/VepAnnotationGeneratorStepParametersValidatorTest.java

```diff
@@ -49,31 +51,63 @@ public void allJobParametersAreValid() throws JobParametersInvalidException, IOE
         appVepPath.setReadable(true);
         inputFasta.setReadable(true);
 
-        JobParametersBuilder jobParametersBuilder = new JobParametersBuilder();
-        jobParametersBuilder.addString(JobParametersNames.APP_VEP_PATH, appVepPath.getCanonicalPath());
-        jobParametersBuilder.addString(JobParametersNames.APP_VEP_CACHE_VERSION, "100_A");
-        jobParametersBuilder.addString(JobParametersNames.APP_VEP_CACHE_PATH, dir.getCanonicalPath());
-        jobParametersBuilder.addString(JobParametersNames.APP_VEP_CACHE_SPECIES, "Human");
-        jobParametersBuilder.addString(JobParametersNames.INPUT_FASTA, inputFasta.getCanonicalPath());
-        jobParametersBuilder.addString(JobParametersNames.APP_VEP_NUMFORKS, "6");
-        jobParametersBuilder.addString(JobParametersNames.OUTPUT_DIR_ANNOTATION, dir.getCanonicalPath());
-        jobParametersBuilder.addString(JobParametersNames.INPUT_STUDY_ID, "inputStudyId");
-        jobParametersBuilder.addString(JobParametersNames.INPUT_VCF_ID, "inputVcfId");
-
-        validator.validate(jobParametersBuilder.toJobParameters());
+        parameters = new TreeMap<>();
+        parameters.put(JobParametersNames.APP_VEP_PATH, new JobParameter(appVepPath.getCanonicalPath()));
+        parameters.put(JobParametersNames.APP_VEP_CACHE_VERSION, new JobParameter("100_A"));
+        parameters.put(JobParametersNames.APP_VEP_CACHE_PATH, new JobParameter(dir.getCanonicalPath()));
+        parameters.put(JobParametersNames.APP_VEP_CACHE_SPECIES, new JobParameter("Human"));
+        parameters.put(JobParametersNames.INPUT_FASTA, new JobParameter(inputFasta.getCanonicalPath()));
+        parameters.put(JobParametersNames.APP_VEP_NUMFORKS, new JobParameter("6"));
+        parameters.put(JobParametersNames.OUTPUT_DIR_ANNOTATION, new JobParameter(dir.getCanonicalPath()));
+        parameters.put(JobParametersNames.INPUT_STUDY_ID, new JobParameter("inputStudyId"));
+        parameters.put(JobParametersNames.INPUT_VCF_ID, new JobParameter("inputVcfId"));
+
+    }
+
+    @Test
+    public void allJobParametersAreValid() throws JobParametersInvalidException, IOException {
+        validator.validate(new JobParameters(parameters));
+    }
+
+    @Test(expected = JobParametersInvalidException.class)
+    public void APP_VEP_PATHIsMissing() throws JobParametersInvalidException, IOException {
```

Comment:
I did this to see what did you guys think, as this way it was copied (at the same time for all the entries with multiple edition) from the parameter in the line below. For the regular capitalization I couldn't find another way that writing one by one. Also, in upper case it's the actual name of the constant.

but well, just a failed experiment